drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c

```diff
@@ -0,0 +1,1271 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @brief File containing QSPI device interface specific definitions for the
+ * Zephyr OS layer of the Wi-Fi driver.
+ */
+
+#define DT_DRV_COMPAT nordic_nrf7002_qspi
```

Comment:
Yes, IIRC, when this was developed QSPI driver was not upstreamed or still WIP, so, this was forked from there and customized, but eventually once support for QSPI driver is complete, we can use that.

Reply:
The Zephyr driver that uses the QSPI peripheral (nrf_qspi_nor) could not be used as it is, because the driver is dedicated for flash chips and, for example, checks JEDEC ID at initialization. That's why it was agreed to copy the needed parts of it here for now and modify as required. Later, when the upstream driver is reworked, this part is supposed to be removed.